<launch>
	<arg name="task" default="goto" />

	<node name="rviz" pkg="rviz" type="rviz" respawn="false" output="screen" args="-d $(find ct_ros_nodes)/rviz/hyaViz.rviz"/>		
	
	<param name="hya_description" textfile="$(find ct_models)/urdf/HyA/HyA.urdf" />
	
	<node pkg="robot_state_publisher" type="robot_state_publisher" name="hya_state_publisher">
 	  <param name="~tf_prefix" value="hya" type="str"/>
      <remap from="robot_description" to="hya_description" />
      <remap from="joint_states" to="current_joint_states" />
    </node>

    <node pkg="robot_state_publisher" type="robot_state_publisher" name="hya_ref_state_publisher">
 		<param name="~tf_prefix" value="hya_ref" type="str"/>
        <remap from="robot_description" to="hya_description" />
        <remap from="joint_states" to="reference_joint_states" />
    </node>
    
    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
    
    <node name="hya_nloc" pkg="ct_ros_nodes" type="HyANLOC" respawn="false" output="screen" >
        <param name="useCodegenModel" value="true" />
        <param name="skip" value="1" />
        <param name="runPerturbedIteration" value="false" />
        <param name="pertScalingFactor" value="0.1" />
       	<param name="workingDirectory" value="$(find ct_ros_nodes)/config/HyA/nloc/$(arg task)" />
    </node>
	
</launch>